File: templates/ruflet_flutter_template/apple_packages/ruflet_apple/Sources/RufletExtensions/RufletFlashlight/Sources/Extension.swift
import RufletEngine

@MainActor
public struct RufletFlashlightExtension: RufletExtension {
  public init() {}

  public var renderedControlTypes: Set<String> { [] }
  public var serviceControlTypes: Set<String> { RufletFlashlight.controlTypes }

  public func createService(for control: RufletControl) -> RufletService? {
    control.type == "Flashlight" ? FlashlightService(control: control) : nil
  }
}

File: templates/ruflet_flutter_template/ruflet_packages/ruflet_flashlight/lib/src/extension.dart
import 'package:ruflet/ruflet.dart';

import 'flashlight.dart';

class Extension extends RufletExtension {
  @override
  RufletService? createService(Control control) {
    switch (control.type) {
      case "Flashlight":
        return FlashlightControl(control: control);
      default:
        return null;
    }
  }
}
